src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -106,7 +106,7 @@ private BoundExpression CompareWithConstant(BoundExpression input, BoundExpressi
                 );
         }
 
-        BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input, LocalSymbol target, bool requiresNullTest)
+        BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input, BoundExpression target, bool requiresNullTest)
```

Comment:
It feels reasonable to assert that it is either a local or a pseudo-local.

---
In reply to: [77735261](https://github.com/dotnet/roslyn/pull/13535#discussion_r77735261) [](ancestors = 77735261,77712197)